lib/client/keystore.go

```diff
@@ -654,3 +654,61 @@ func initKeysDir(dirPath string) (string, error) {
 
 	return dirPath, nil
 }
+
+type usernameProxyPair struct {
+	username string
+	proxy    string
+}
+
+// MemLocalKeyStore is a wrapper around any LocalKeyStore
+// with the ability to keep newly added keys in memory instead
+// of writing them to the underlying key store.
+type MemLocalKeyStore struct {
+	LocalKeyStore
+	saveNewKeys bool
```

Comment:
nit: no need for this to be configurable
`MemLocalKeyStore` should only be used when keys should not be stored in the real keystore
and that means `AddKey/GetKey/DeleteKey/DeleteKeys` should only deal with `inMem`, not forward calls to the inner `LocalKeyStore`

Reply:
Ah, misunderstood some parts of the rfc then and how tsh works. I thought it still needed to load some stuff from the FS.